From 1c9b976d7657c98fd76a27c841238d70fadf8e46 Mon Sep 17 00:00:00 2001
From: Chuanxiao Dong <chuanxiao.dong@intel.com>
Date: Wed, 5 Sep 2012 13:50:53 +0800
Subject: [PORT FROM MAIN] vold: use SDCARD_RW for secondary storage

BZ: 55162

Use SDCARD_RW permission to the secondary storage, so that the 3rd
party application can use it

Change-Id: Ie8c9b76e1c42f1782c5f50e87f98910668fc5951
Orig-Change-Id: I2eb5a914f1fe2821f6f37d5f55201fb2b861faca
Signed-off-by: Chuanxiao Dong <chuanxiao.dong@intel.com>
Reviewed-on: http://android.intel.com:8080/75936
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Sun, Jianhua <jianhua.sun@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 Volume.cpp |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/Volume.cpp b/Volume.cpp
index c676838..5cd8fb6 100644
--- a/Volume.cpp
+++ b/Volume.cpp
@@ -432,7 +432,12 @@ int Volume::mountVol() {
             gid = AID_SDCARD_RW;
         } else {
             // For secondary external storage we keep things locked up.
-            gid = AID_MEDIA_RW;
+            /*
+             * Google set gid = AID_MEDIA_RW to keep things locked up.
+             * But we want the 3rd party application like file manager
+             * can still have the right to access it. So keep use AID_SDCARD_RW.
+             */
+            gid = AID_SDCARD_RW;
         }
         if (Fat::doMount(devicePath, "/mnt/secure/staging", false, false, false,
                 AID_SYSTEM, gid, 0702, true)) {
-- 
1.7.4.1

